app/experimenter/nimbus-ui/README.md

```diff
@@ -89,11 +89,35 @@ export interface GetExperimentOverviews {
 
 This app has an [`AppErrorBoundary`](./src/components/AppErrorBoundary/index.tsx) that will capture any uncaught error that occurs and report them to Sentry. This acts as our last line of defense, but ideally we are able to handle errors before they get to this stage. As well, errors that we wish to display to the user should generally be handled in a consistent fashion.
 
-There are two types of errors to account for:
+### General errors
+
+General errors can occur when an operation fails altogether. This could be for any reason, such as when a network request fails, or a bad code path raises an exception. For these errors you'll use React Bootstrap's [`Alert`](https://react-bootstrap.github.io/components/alerts/) components.
+
+For example, if error occurs while saving a record:
+
+```tsx
+<Alert variant="warning">Sorry, there was a problem.</Alert>
+```
+
+_Note:_ displaying technical error messages to the user may not be very helpful, so it's generally advised to either provide a [generic message](./src/lib/constants.ts) or instructions on what to do.
+
+## Form Validation
+
+TODO: Replace storybook links.
+
+There are three kinds of validation that need to be applied to forms: client-side, server-side, and "required for launch". For a visual guide to the interaction UX, see this STORYBOOK LINK
+
+#### Client-side validation errors
 
-### Validation errors
+Client-side validation should be used to prevent invalid mutations from being attempted (i.e. the wrong type or fields that cannot be `null` in the database). It should NOT be used to validate fields that are required for launch or to duplicate server-side validation.
 
-These occur when you try to create or modify a record and the server tells us that one or more fields provided incorrect data. For these you should use React Bootstrap's [`Form.Control.Feedback`](https://react-bootstrap.github.io/components/forms/#form-control-feedback-props) component in combination with other form components.
+#### Required for launch validation
+
+Some fields like `publicDescription` are allowed to be `null` while someone is editing a draft, but must be set before the experiment can move to the `review` status. You should NOT mark these fields `required`, but rather add help text that indicates they must be filled out before launch.
```

Comment:
The functionality is not there yet, but IIRC eventually _all_ fields will be optional until review (instead of just some)

Reply:
Yeah I should clarify – I wrote this as a "it should be this way"

Reply:
Looks like `name` for experiment and branches is not allowed to be `null` in the DB (which makes sense),  but yeah looks like those are the only ones – or are we planning on changing that @jaredlockhart? 

 I should update this to "most fields are allowed to be `null` when editing a draft...."

Reply:
Per [this thread](https://mozilla.slack.com/archives/CANCHH6R1/p1608157109107400?thread_ts=1608153103.100700&cid=CANCHH6R1) it sounds like it should be fine as far as UI goes